backend/src/data/repositories/diagnosis.repository.ts

```diff
@@ -0,0 +1,17 @@
+import { DiagnosisModel } from '../models';
+import { IDiagnosis } from '~/common/interfaces';
+
+class Diagnosis {
+  public create(diagnosis: IDiagnosis): Promise<IDiagnosis> {
+    return DiagnosisModel.create(diagnosis);
+  }
+
+  public getAllByUserId(userId: string): Promise<IDiagnosis[]> {
+    return DiagnosisModel.findAll({
+      where: { userId },
+      order: [['createdAt', 'DESC']],
```

Comment:
```suggestion
      order: [['createdAt', 'DESC']],
```
User enum key here

Create SortType enum